go/api/response_handlers.go

```diff
@@ -280,6 +280,25 @@ func handleBooleanResponse(response *C.struct_CommandResponse) (Result[bool], er
 	return CreateBoolResult(bool(response.bool_value)), nil
 }
 
+func handleBooleanArrayResponse(response *C.struct_CommandResponse) ([]Result[bool], error) {
```

Comment:
It makes an array of `Result`s, but why not to do a `Result` with an array?
I understand that you copied that code from `handleLongArrayResponse`. Please discuss this with the team and update both if needed.
My only concert is ease of usage - call `.Value()` once and get a plain array of plain bools is easier than calling `.Value()` for every element in the array.

Reply:
We have wrapped every individual inside Result because we want to convey the information of every value to the client separately that whether the individual value was `nil` or not. This is because `nil` value can't be assigned to the primitive types in GO and there is no easy way to pass the `nil` info to the client.

Reply:
Hi @janhavigupta007, thank you for your explaination. I understand that the purpose of creating such a struct to begin with is to add nullability to primitive types of GO. I'm new to GO but an alternative I can think of is to use pointers. Have we considered this option?

Reply:
We had considered this option but we went ahead with struct approach to provide clear information to the client that the value is missing.